### PR TITLE
max listeners

### PR DESCRIPTION
**Pull request by ash-developer**
made fixes:
- maxListeners with value 0 at initial config doesn't doesn't override default value 10
- doesn't warn about 2 listeners with maxListeners value 1
improved performance

### CODE REVIEW

File: lib/eventemitter2.js [hunk 1]
```diff
@@ -182,32 +194,20 @@
         if (!tree._listeners) {
           tree._listeners = listener;
         }
-        else if(typeof tree._listeners === 'function') {
-          tree._listeners = [tree._listeners, listener];
-        }
-        else if (isArray(tree._listeners)) {
+        else {
+          if (typeof tree._listeners === 'function') {
+            tree._listeners = [tree._listeners];
```
Comment (RangerMauve) [inline lib/eventemitter2.js]: Why did you remove the second element in this array?

File: lib/eventemitter2.js [hunk 2]
```diff
@@ -670,15 +660,14 @@
         leaf._listeners = null;
       }
     }
-    else {
-      if (!this._events || !this._events[type]) return this;
+    else if (this._events) {
```
Comment (RangerMauve) [inline lib/eventemitter2.js]: What is this change for?

File: lib/eventemitter2.js (hunk 1 above)
Comment (ash-developer) [inline lib/eventemitter2.js]: i didn't, if type is function then i transform it to array and after that make push
it is needed for check memory leak when maxListeners is 1 and we add second listener

File: lib/eventemitter2.js (hunk 2 above)
Comment (ash-developer) [inline lib/eventemitter2.js]: i think it's more clear way to remove listeners
if there is some issues with it i can rollback it

Comment (RangerMauve) [inline lib/eventemitter2.js]: Should you add a `this.events[type]`?

File: lib/eventemitter2.js (hunk 1 above)
Comment (RangerMauve) [inline lib/eventemitter2.js]: Makes sense, thanks for clarifying!

File: lib/eventemitter2.js (hunk 2 above)
Comment (RangerMauve) [inline lib/eventemitter2.js]: I guess it doesn't matter, actually.

**Comment by RangerMauve** [comment]
Thanks for the PR! It's out in 2.1.3
